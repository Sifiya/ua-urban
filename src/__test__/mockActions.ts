import * as actions from '@/app/api/actions';
import * as authApi from '@/app/api/auth.api';

jest.mock('@/app/api/actions', () => ({
  addWordWithDefinition: jest.fn(),
  getAllWords: jest.fn(),
  addDefinition: jest.fn(),
}));

export const mockAddWordWithDefinition = actions.addWordWithDefinition as jest.Mock;
export const mockGetAllWords = actions.getAllWords as jest.Mock;
export const mockAddDefinition = actions.addDefinition as jest.Mock;

jest.mock('@/app/api/auth.api', () => ({
  getUser: jest.fn(),
}));

export const mockGetUser = authApi.getUser as jest.Mock;
