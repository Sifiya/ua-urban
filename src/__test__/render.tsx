import React, { ReactNode } from 'react';
import { render as libRender } from '@testing-library/react';
import { Providers } from '@/components/Providers';

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <Providers isTest>{children}</Providers>;
};

export const render = (ui: ReactNode) => {
  return libRender(ui, {
    wrapper: Wrapper,
  });
};
