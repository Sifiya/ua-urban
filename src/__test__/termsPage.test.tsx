import React from 'react';
import { screen } from '@testing-library/react';
import { render } from '@/__test__/render';
import TermsPage, { metadata } from '../app/terms/page';

describe('TermsPage', () => {
  test('should render', async () => {
    const { container } = render(<TermsPage />);
    expect(screen.getByRole('heading', { name: 'Умови користування' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    expect(metadata).toMatchInlineSnapshot(`
      {
        "title": "UA Urban - Умови користування сайтом",
      }
    `);       
  });
});
