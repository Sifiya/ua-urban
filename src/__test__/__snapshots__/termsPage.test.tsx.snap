// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsPage should render 1`] = `
<div>
  <section>
    <h1
      class="
      scroll-m-20 text-4xl font-extrabold tracking-tight lg:text-5xl
      undefined
    "
    >
      Умови користування
    </h1>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Ласкаво просимо на сайт «UA Urban»! Нижче наведені умови користування, які регулюють ваш доступ та використання нашого сервісу. Будь ласка, уважно прочитайте цей документ перед використанням сайту.
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Прийняття умов.
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Користуючись нашим сайтом, ви погоджуєтеся з цими умовами користування в повному обсязі. Якщо ви не згодні з будь-якою частиною цих умов, будь ласка, не використовуйте наш сервіс.
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Мета сайту
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Основною метою є збір та поширення коректної та корисної інформації про сучасну українську мову в формі доступній необмеженому колу осіб.
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Вміст
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Користувачі мають можливість додавати нові слова та визначення до словника, а також голосувати за якість визначень. Додаючи слова, визначення та інший контент до сайту ви погоджуєтеся з тим, що він може бути вільно використаний адміністрацією сайту або третіми особами в будь-якій формі з некомерційною або комерційною метою. Адміністрація сайту не несе відповідальності за контент, створений користувачами.
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Обмеження
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Веб-сайт не підходить для всіх аудиторій. Його зміст часто представлений у грубій і прямій формі, що може бути для когось образливим. Якщо ви не вважаєте себе відповідним користувачем або може бути ображені вмістом сайту, будь ласка, не відвідуйте веб-сайт.
    </p>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Сайт не призначений для дітей молодше 16 років. Використовуючи веб-сайт, ви підтверджуєте, що вам виповнилося 16 років.
    </p>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Ви не можете використовувати сайт для будь-якої незаконної діяльності або порушення законів у вашій юрисдикції.Ви не можете використовувати сайт для отримання доступу до несанкціонованої інформації.
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Модерація
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Ми робимо все можливе, щоб забезпечити якість та безпеку сайту. Адміністрація сайту здійснює модерацію контенту та може вилучати будь-який вміст, що порушує чинник законодавство України або протирічить меті та ідеї сайту. Вилучення відбувається без попереднього повідомлення.
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Персональні дані
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Деякі функції нашого сайту доступні тільки для зареєстрованих користувачів. Дані, що ви надаєте ми використовуємо відповідно до 
      <svg
        class="lucide lucide-link "
        fill="none"
        height="24"
        href="/privacy"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"
        />
        <path
          d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"
        />
        політики конфіденційності
      </svg>
      .
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Обмеження відповідальності
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Веб-сайт надається за принципом "як є" і "як доступно", без будь-яких явних або неявних
        гарантій будь-якого виду. Ми прямо відмовляємося від будь-яких неявних гарантій придатності для продажу,
        гарантій придатності для певної мети, гарантій непорушення прав або будь-яких гарантій того,
        що використання веб-сайту буде безперервним або безпомилковим.
        Ви берете на себе повну відповідальність і ризик за використання вами веб-сайту.
    </p>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      В повній мірі, дозволеній чинним законодавством, за жодних обставин і за жодної правової теорії компанія, її ліцензіари або постачальники не несуть відповідальності перед вами або будь-якою іншою особою за будь-які непрямі, спеціальні, випадкові або побічні збитки будь-якого виду.
    </p>
    <h3
      class="
        mt-8 scroll-m-20 text-2xl font-semibold tracking-tight
        undefined
      "
    >
      Зміни умов
    </h3>
    <p
      class="leading-7 [&:not(:first-child)]:mt-6 undefined"
    >
      Ми залишаємо за собою право змінювати ці умови користування в будь-який час без попереднього повідомлення. Будь ласка, регулярно перевіряйте цей документ на наявність змін.
    </p>
  </section>
</div>
`;
